src/Reopt/TypeInference/ConstraintGen.hs

```diff
@@ -129,165 +131,226 @@ newtype Warning = Warning String
 instance Show Warning where
   show (Warning w) = w
 
-type PhiType arch = Map (Some (FnPhiVar arch)) IType
+-- | Context available throughout all of the constraint generation
```

Comment:
I don't think splitting up the context like this is worth the extra complexity: was there a good reason why you made the change?

Reply:
I started moving things around and running into errors because some things were not initialized by the time some operations were called.
With this change, it is no longer possible to be in a state where you don't have the necessary available data when needed, and it also goes out of scope when you move to another block to prevent uses of stale values.

Before, if you added fields to the context data type, you had to be extremely careful about all the places where some fields needed to be filled or overwritten because their old value was no longer relevant.